### PR TITLE
Staging for version increment automation

### PR DESCRIPTION
**Pull request by prudhvigodithi**
Signed-off-by: pgodithi <pgodithi@amazon.com>

### Description
This is the staging PR for to execute workflows that auto raise the version increment PR's
Example: https://github.com/prudhvigodithi/security/pull/2
* Category: Enhancement

### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/1375
From solution: https://github.com/opensearch-project/opensearch-build/issues/1375#issuecomment-1159476686
Related issue: https://github.com/opensearch-project/security/issues/1930

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Example: https://github.com/prudhvigodithi/security/pull/2

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Referred old 2.x version increment PR: https://github.com/opensearch-project/security/pull/1881
@dblock  @bbarani

**Review by peternied — CHANGES_REQUESTED** [review]
Before we merge this change should have a test to ensure its functionality.

File: build.gradle [hunk 1]
```diff
@@ -486,3 +486,20 @@ afterEvaluate {
         tasks = ['build', 'buildRpm', 'buildDeb']
     }
 }
+
+// versionIncrement: Task to auto increment to the next development iteration
+task versionIncrement {
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply build.gradle:491]: Added @peternied can you please check.
Thank you

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,6 +125,8 @@ jobs:
 
     - run: ./gradlew clean assemble -Dbuild.version_qualifier=${{ env.TEST_QUALIFIER }} && test -s ./build/distributions/opensearch-security-${{ env.SECURITY_PLUGIN_VERSION_ONLY_NUMBER }}-${{ env.TEST_QUALIFIER }}-SNAPSHOT.zip
 
+    - run: ./gradlew clean versionIncrement -DnewVersion=3.0.0-SNAPSHOT
```
Comment (peternied) [inline .github/workflows/ci.yml:128]: This executes the command, but doesn't verify any outcome.  

What do you think about adding a execution of the gradle task `assemble`, check the artifact is version X, then run the `versionIncrement` task, then `assemble` and check its version X+1

**Comment by prudhvigodithi** [comment]
Renaming to task updateVersion following https://github.com/opensearch-project/opensearch-plugin-template-java/pull/32.

**Review by DarshitChanpura — CHANGES_REQUESTED** [review]
This looks good. I agree with Peter's ask on testing to verify whether this change would work, if it is possible.

**Comment by prudhvigodithi** [comment]
> @prudhvigodithi Do you need any help with this?

Hey @peternied I should be pushing the requested change by end of today, just got pulled away from this :)

**Comment by prudhvigodithi** [comment]
Hey @peternied I have updated the CI workflow running the task `updateVersion` with an increment of `majorVersion + 1`. Please check
Thank you 
https://github.com/opensearch-project/security/runs/7512662131?check_suite_focus=true
@bbarani

**Comment by bbarani** [comment]
@davidlago @peternied @DarshitChanpura  Can you please review this PR soon?

**Review by peternied — APPROVED** [review]
Thanks!

**Review by DarshitChanpura — COMMENTED** [review]
Thank you @prudhvigodithi !

**Review by DarshitChanpura — APPROVED** [review]
Thank you @prudhvigodithi !

**Comment by prudhvigodithi** [comment]
Hey @peternied I see the following PR is closed https://github.com/opensearch-project/security/pull/1933, but this change is not backported to 1.x (1.3) branches, can you please take a look?

Thank you

**Comment by peternied** [comment]
@prudhvigodithi Could you double check https://github.com/opensearch-project/security/pull/2029 and make sure it will work as expected?

**Pull request merged**